components/uaepass-authenticator/src/main/java/org/wso2/carbon/identity/authenticator/uaepass/UAEPassAuthenticator.java

```diff
@@ -0,0 +1,958 @@
+                    setGrantType(GrantType.AUTHORIZATION_CODE).setClientId(clientId).setClientSecret(clientSecret).
+                    setRedirectURI(callbackUrl).setCode(authzResponse.getCode()).buildBodyMessage();
+
+            if (accessTokenRequest != null) {
+                String serverURL = ServiceURLBuilder.create().build().getAbsolutePublicURL();
+                accessTokenRequest.addHeader(UAEPassAuthenticatorConstants.UAE.HTTP_ORIGIN_HEADER, serverURL);
+            }
+
+        } catch (OAuthSystemException e) {
+            LOG.error("Unable to build the request with request's body attributes.");
+            throw new UAEPassAuthnFailedException("Exception while building access token request "
+                    + "with the request body", e);
+        } catch (URLBuilderException e) {
+            LOG.error("Unable to identify common-auth URL on browser.");
+            throw new UAEPassAuthnFailedException("Error occurred while extracting the absolute public " +
+                    "URL from browser.", e);
+        }
+
+        return accessTokenRequest;
+    }
+
+    /**
+     * Returns the OAuth type response to the back channel.
+     *
+     * @param oAuthClient                   OAuth client object received to the authenticator.
+     * @param accessRequest                 OAuth client request received by the authenticator.
+     * @return OAuthClientResponse          Returns the OAuth client response from the authenticator.
+     * @throws UAEPassAuthnFailedException  UAEPassAuthnFailedException will throw to the processAuthenticationResponse.
+     */
+    public OAuthClientResponse getOAuthResponse(OAuthClient oAuthClient, OAuthClientRequest accessRequest)
+            throws UAEPassAuthnFailedException {
+
+        OAuthClientResponse oAuthResponse = null;
+        try {
+            oAuthResponse = oAuthClient.accessToken(accessRequest);
+        } catch (OAuthSystemException | OAuthProblemException e) {
+            LOG.error("UAEPass OAuth client response failed.");
+            throw new UAEPassAuthnFailedException("Unable to return OAuth client response.", e);
+        }
+
+        return oAuthResponse;
+    }
+
+    /**
+     * This method specifies the OIDC to the state parameter.
+     *
+     * @param request  The request that is received by the authenticator.
+     * @return String  Returns the login type of the authenticator.
+     */
+    public String getLoginType(HttpServletRequest request) {
+
+        String state = request.getParameter(UAEPassAuthenticatorConstants.UAE.OAUTH2_PARAM_STATE);
+        if (StringUtils.isNotBlank(state)) {
+            String[] stateElements = state.split(",");
+            if (stateElements.length > 1) {
+                return stateElements[1];
+            }
+        }
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Empty split elements in state");
+            LOG.debug("Received request path info : " + request.getPathInfo());
```

Comment:
Can we merge these two debug logs?

Reply:
Fixed with : [e41125306](https://github.com/wso2-extensions/identity-outbound-auth-uaepass/pull/1/commits/e41125306054f4b7345ef374ad89a9ab2fdfea52)